### PR TITLE
Cleanup TopologyManager and update policy.Merge()

### PR DESCRIPTION
**Pull request by klueska**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug

/kind cleanup

> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
This PR cleans up the logic in the TopologyManager pod Admit hander. As part of this, it simplifies the interface for its underlying policy.Merge() calls. 

It is best reviewed commit-by-commit.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/cm/topologymanager/topology_manager.go [hunk 1]
```diff
@@ -194,10 +194,15 @@ func (m *manager) AddContainer(pod *v1.Pod, containerID string) error {
 }
 
 func (m *manager) RemoveContainer(containerID string) error {
+	klog.Infof("[topologymanager] RemoveContainer - Container ID: %v", containerID)
+
 	podUIDString := m.podMap[containerID]
-	delete(m.podTopologyHints, podUIDString)
 	delete(m.podMap, containerID)
-	klog.Infof("[topologymanager] RemoveContainer - Container ID: %v podTopologyHints: %v", containerID, m.podTopologyHints)
+	delete(m.podTopologyHints[podUIDString], containerID)
```
Comment (ipuustin) [inline pkg/kubelet/cm/topologymanager/topology_manager.go]: The old implementation allowed `RemoveContainer` to be called even when the `m.podTopologyHints[podUIDString]` didn't exist. Should we check here for existence before delete?

**Comment by klueska** [comment]
Yes. We should check for that. That was an oversight on my part.

**Pull request merged**